lizard_connector/connector.py

```diff
@@ -0,0 +1,342 @@
+    def header(self):
+        """
+        The header with credentials for the api.
+        """
+        if self.use_header:
+            return {
+                "username": self.username,
+                "password": self.password
+            }
+        return {}
+
+
+def parse_element(json_, element):
+    """
+    Get a list of a certain element from the root of the results attribute.
+
+    :param json: json from the Lizard api parsed into a dictionary.
+    :param element: the element you wish to get.
+    :return: A list of all elements in the root of the results attribute.
+    """
+    return [x[element] for x in json_]
+
+
+def parse_uuid(json_, endpoint=None):
+    """
+    Get a list of a certain element from the root of the results attribute.
+
+    :param json: json from the Lizard api parsed into a dictionary.
+    :param endpoint: endpoint you wish to query.
+    :return: A list of all uuid elements in the root of the results attribute.
+    """
+    uuid = 'unique_id' if endpoint == 'organisations' else 'uuid'
+    return parse_element(json_, uuid)
+
+
+class Endpoint(object):
+    max_results = 1000
+    endpoint = ""
+
+    def __init__(self, base, **kwargs):
+        """
+        :param base: lizard-nxt url.
+        """
+        self.fetcher = Fetcher(
+            endpoint=self.endpoint,
+            max_results=self.max_results,
+            **kwargs
+        )
+        base = base.strip(r'/')
+        if not base.startswith('http'):
+            base = 'https://' + base
+        base = urllib.parse.urljoin(base, 'api/v2')
+        self.base_url = urllib.parse.urljoin(base, self.endpoint) + "/"
+
+    def get(self, *querydicts, **queries):
+        """
+        Query the api at this endpoint.
+        For possible queries see: https://nxt.staging.lizard.net/doc/api.html
+        Stores the api-response as a dict in the results attribute.
+
+        :param querydicts: all key valuepairs from dictionaries are used as
+            queries.
+        :param queries: all keyword arguments are used as queries.
+        :return:
+        """
+        q = QueryDictionary(page_size=self.max_results)
+        q.update(*querydicts, **queries)
+        query = "?" + urllib.parse.urlencode(q)
+        url = urllib.parse.urljoin(self.base_url, query)
+        return self.fetcher.get(url)
+
+    def post(self, uuid, data):
+        """
+        POST data to the api at this endpoint.
+        :param uuid: UUID of the object in the database you wish to store
+                     data to.
+        :param data: Dictionary with the data to post to the api
+        """
+        self.fetcher.post(self.base_url, uuid, data)
+
+    @property
+    def paginated(self):
+        """
+        Indicates whether this object is paginated (i.e. other pages exist).
+        """
+        return bool(self.fetcher.next_url)
+
+    @property
+    def count(self):
+        """
+        Number of results.
+        """
+        return self.fetcher.count
```

Comment:
Hm. Ook twijfelachtig. Kan misschien wel, maar de twee classes lijken wat kunstmatig gescheiden te zijn. Is dat misschien omdat je te slim wilt zijn met iterators? Kan je ze samenvoegen?